sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -592,6 +594,42 @@ def __init__(self, special_parameter):
                                   check_estimator, MyEstimator)
 
 
+class LRXFailTags(LogisticRegression):
+    def _more_tags(self):
+        return {"_xfail_checks": {
+            "check_complex_data": "This is a bad check"}
+        }
+
+
+def test_check_estimator_xfail_tag_raises():
+    # When a _xfail_checks check passes, raise an error stating that the test
+    # passes and can be removed from the tag
```

Comment:
My understanding was that we dont't care about that scenario because we don't want to reimplement pytest's logic?

Reply:
I changed it to issuing a warning.

From an implementation point of view, it only takes 2 lines to implement this logic in `_check_warns_on_fail`. I would not want to keep silent about this, since a developer would not know if a xfail check is now passing.